src/libcore/tests/num/dec2flt/mod.rs

```diff
@@ -99,6 +99,8 @@ fn fast_path_correct() {
 fn lonely_dot() {
     assert!(".".parse::<f32>().is_err());
     assert!(".".parse::<f64>().is_err());
+    assert!(".e0".parse::<f32>().is_err());
+    assert!(".e0".parse::<f64>().is_err());
```

Comment:
This doesn't really match the test name IMO. Better create a new test with an appropriate name (or `issue_40654` if you, like me, can't think of a good succinct name).